src/GovUk.Education.ExploreEducationStatistics.Content.Services/GlossaryService.cs

```diff
@@ -0,0 +1,57 @@
+#nullable enable
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Common.Cache;
+using GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using GovUk.Education.ExploreEducationStatistics.Content.Services.Cache;
+using GovUk.Education.ExploreEducationStatistics.Content.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Content.Services.ViewModels;
+
+namespace GovUk.Education.ExploreEducationStatistics.Content.Services
+{
+    public class GlossaryService : IGlossaryService
+    {
+        private readonly ContentDbContext _contentDbContext;
+
+        public GlossaryService(ContentDbContext contentDbContext)
+        {
+            _contentDbContext = contentDbContext;
+        }
+
+        [BlobCache(typeof(GlossaryCacheKey))]
+        public async Task<List<GlossaryCategoryViewModel>> GetAllGlossaryEntries()
+        {
+            var categories = "ABCDEFGHIJKLMNOPQRSTUVWXYZ".ToCharArray();
+            var categoryDictionary = categories.ToDictionary(
+                c => c.ToString(),
+                c => new GlossaryCategoryViewModel
+                {
+                    Heading = c.ToString(),
+                    Entries = new List<GlossaryEntryViewModel>(),
+                });
+
+            var allGlossaryEntries = _contentDbContext.GlossaryEntries
+                .ToList();
+
+            allGlossaryEntries
+                .OrderBy(e => e.Title)
```

Comment:
I think I would have done the ordering in the database query above but it probably makes little difference.

Reply:
I did try that but it gave me an error so I switched to doing the ordering here - but I just had a second attempt at it - I needed to add an initial `ToAsyncEnumerable` to make it work. So I'm ordering on the database query now.